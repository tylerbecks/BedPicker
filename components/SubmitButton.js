import React from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";

const styles = StyleSheet.create({
  button: {
    alignItems: "center",
    backgroundColor: "#21ba45",
    borderRadius: 100,
    justifyContent: "center",
    padding: 10,
    height: 80,
    margin: 10
  },
  disabled: {
    opacity: 0.4
  },
  text: {
    color: "white",
    fontWeight: "bold",
    fontSize: 20
  }
});

const SubmitButton = ({ onPress, disabled, text }) => (
  <TouchableOpacity
    activeOpacity={disabled ? 1 : 0.7}
    style={disabled ? { ...styles.button, ...styles.disabled } : styles.button}
    onPress={onPress}
    disabled={disabled}
  >
    <Text style={styles.text}>{text}</Text>
  </TouchableOpacity>
);

export default SubmitButton;
