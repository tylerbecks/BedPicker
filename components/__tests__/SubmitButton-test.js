import 'react-native';
import React from 'react';
import SubmitButton from '../SubmitButton';
import renderer from 'react-test-renderer';

it('renders with blue background when selected', () => {
  expect(
    renderer.create(<SubmitButton onPress={jest.fn()} />).toJSON()
  ).toMatchSnapshot();
});
