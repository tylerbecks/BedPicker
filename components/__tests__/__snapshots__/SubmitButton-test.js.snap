// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with blue background when selected 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#21ba45",
      "borderRadius": 100,
      "height": 80,
      "justifyContent": "center",
      "margin": 10,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Submit
  </Text>
</View>
`;
